lib/KDateRange/KDateCalendar.vue

```diff
@@ -0,0 +1,475 @@
+        activeMonth: new Date().getMonth() - 1,
+        activeYearStart: new Date().getFullYear(),
+        activeYearEnd: new Date().getFullYear(),
+      };
+    },
+    computed: {
+      /**
+       * returns the day of week of the beginning of the left side calendar month
+       */
+      activeMonthDay() {
+        return new Date(this.activeYearStart, this.activeMonth, 1).getDay();
+      },
+      /**
+       * returns the day of week of the beginnning of the right side month
+       */
+      nextActiveMonthDay() {
+        return new Date(this.activeYearEnd, this.nextActiveMonth, 1).getDay();
+      },
+      /**
+       * returns the last day of the month on the left side calendar month
+       */
+      activeMonthDate() {
+        return new Date(this.activeYearEnd, this.nextActiveMonth, 0).getDate();
+      },
+      /**
+       * returns the last day of the month on the right side calendar month
+       */
+      nextActiveMonthDate() {
+        return new Date(this.activeYearEnd, this.activeMonth + 2, 0).getDate();
+      },
+      /**
+       * returns the next month that comes after the current active month
+       */
+      nextActiveMonth() {
+        return this.activeMonth >= 11 ? 0 : this.activeMonth + 1;
+      },
+      buttonStyles() {
+        return {
+          '--selected-button-bg-color': this.$themeBrand.secondary.v_50,
+          '--disabled-button-color': this.$themePalette.grey.v_300,
+        };
+      },
+    },
+    watch: {
+      nextActiveMonth(value) {
+        if (value === 0) this.activeYearEnd = this.activeYearStart + 1;
+      },
+    },
+    created() {
+      if (this.activeMonth === 11) this.activeYearEnd = this.activeYearStart + 1;
+    },
+    methods: {
+      /**
+       * returns the index number within month of where day should be placed
+       */
+      getDayIndexInMonth(weekIndex, dayInWeekIndex, activeMonthDay) {
+        const date = this.numOfDays * (weekIndex - 1) + dayInWeekIndex;
+        return date - activeMonthDay;
+      },
+      /**
+       * returns placement where day should be placed on calendar
+       */
+      getDayCell(weekIndex, dayInWeekIndex, activeMonthDay, activeMonthDate) {
+        const result = this.getDayIndexInMonth(weekIndex, dayInWeekIndex, activeMonthDay);
+        // bound by > 0 and < last day of month
+        return result > 0 && result <= activeMonthDate ? result : null;
+      },
+      /**
+       * update end and start date of dateRange Object
+       */
+      getNewDateRange(result, activeMonth, activeYear) {
+        const newData = {};
+        let key = 'start';
+        if (!this.isFirstChoice) {
+          key = 'end';
+        } else {
+          newData['end'] = null;
+        }
+        const resultDate = new Date(activeYear, activeMonth, result);
+        if (!this.isFirstChoice && resultDate < this.dateRange.start) {
+          this.isFirstChoice = false;
+          return { start: resultDate };
+        }
```

Comment:
Since this `if` block returns early, and doesn't depend on the code above (that I can see), would be good to move it to the beginning of the function.